### DISCUSSION TITLE
Root file not found at any remote fs

### DISCUSSION THREAD

**Discussion by JulesVandenbroeck** [columnflow/columnflow#387] [Q&A]
Hi everyone,

I have received trouble recently with opening certain root-files in cf (that previously worked). The file that cf mentions in the error (below) does however exist and can be opened using `uproot.open("root://cms-xrd-global.cern.ch//"+file_path)`. In addition files from the same dataset are able to be opened. Since I can open the file by myself I assume this is not an issue of my proxy or the file itself, therefore this discussion.

The error:
`Exception: LFN /store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-170to300_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/250000/E9F238E8-B0E2-BE4B-B205-8730ACBFFBBC.root not found at any remote fs ['wlcg_fs_t2b_redirector', 'wlcg_fs_infn_redirector', 'wlcg_fs_global_redirector']`

wlcg_fs_global_redirector in law.cfg:

```
[wlcg_fs_global_redirector]

base: root://cms-xrd-global.cern.ch/
use_cache: $CF_WLCG_USE_CACHE
cache_root: $CF_WLCG_CACHE_ROOT
cache_cleanup: $CF_WLCG_CACHE_CLEANUP
cache_max_size: 15GB
cache_global_lock: True
cache_mtime_patience: -1
```

**Comment by riga** [comment]
Hi @JulesVandenbroeck ,

could you paste logs of the `GetDatasetLFNs` task as well? It prints a couple lines when it tries to  reach the redirectors / fs's that are helpful for debugging :)

**Reply by JulesVandenbroeck** [reply]
Do you mean the output of the GetDatasetLFNs task, I placed it below. If you mean there is a log-file output somewhere I am unsure where it would be stored?

```
=============================== entering sandbox ===============================
task   : cf.GetDatasetLFNs_cffakerates_anal_e18_qcd_em_pt170to30_04cdf8f00f
sandbox: bash::/cvmfs/cms.cern.ch/cmsset_default.sh
================================================================================

INFO: luigi-interface - Done scheduling tasks
INFO: luigi-interface - Running Worker with 1 processes
INFO: luigi-interface - [pid 28134] Worker Worker(salt=9062942062, workers=1, host=m3.iihe.ac.be, username=juvanden, pid=28086) running   cf.GetDatasetLFNs(source_path=NO_STR, analysis=cffakerates.analysis.cffakerates.cffakerates, config=e18, shift=nominal, local_shift=nominal, dataset=qcd_em_pt170to300_pythia, replicas=5)
INFO: cf.GetDatasetLFNs_cffakerates_anal_e18_qcd_em_pt170to30_04cdf8f00f - get lfns for dataset key /QCD_Pt-170to300_EMEnriched_TuneCP5_13TeV-pythia8/RunIISummer20UL18NanoAODv9-106X_upgrade2018_realistic_v16_L1v1-v2/NANOAODSIM via dasgoclient
INFO: cf.GetDatasetLFNs_cffakerates_anal_e18_qcd_em_pt170to30_04cdf8f00f - found 5 lfn(s) for dataset qcd_em_pt170to300_pythia
uploaded lfns_96d2dad773.0.json
uploaded lfns_96d2dad773.1.json
uploaded lfns_96d2dad773.2.json
uploaded lfns_96d2dad773.3.json
uploaded lfns_96d2dad773.4.json
INFO: luigi-interface - [pid 28134] Worker Worker(salt=9062942062, workers=1, host=m3.iihe.ac.be, username=juvanden, pid=28086) done      cf.GetDatasetLFNs(source_path=NO_STR, analysis=cffakerates.analysis.cffakerates.cffakerates, config=e18, shift=nominal, local_shift=nominal, dataset=qcd_em_pt170to300_pythia, replicas=5)
INFO: luigi-interface - This progress looks :) because there were no failed tasks or missing dependencies

=============================== leaving sandbox ================================
task   : cf.GetDatasetLFNs_cffakerates_anal_e18_qcd_em_pt170to30_04cdf8f00f
sandbox: bash::/cvmfs/cms.cern.ch/cmsset_default.sh
================================================================================

```

**Reply by riga** [reply]
Sorry, my bad. I meant the log (and yes, just the lines it prints in your terminal) of the task that tries to access the lfn, so I guess either `{Calibrate|Select|Reduce}Events`.

**Reply by JulesVandenbroeck** [reply]
This is a log-file from running cf.ReduceEvents on another dataset with the same issue ([stdall_1To2.txt](https://github.com/columnflow/columnflow/files/13991109/stdall_1To2.txt)), I assume the important lines are:
```
INFO: columnflow.tasks.external - file /store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-80to120_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/2430000/3CD20F5B-ED95-F14F-9A9D-CB69309D2380.root does not exist at fs wlcg_fs_t2b_redirector
INFO: columnflow.tasks.external - file /store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-80to120_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/2430000/3CD20F5B-ED95-F14F-9A9D-CB69309D2380.root does exist at fs wlcg_fs_infn_redirector
INFO: columnflow.tasks.external - file /store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-80to120_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/2430000/3CD20F5B-ED95-F14F-9A9D-CB69309D2380.root does not exist at fs wlcg_fs_global_redirector
```

The third line also confuses me as I just ran the following command without any issues: `f = uproot.open("root://cms-xrd-global.cern.ch///store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-80to120_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/2430000/3CD20F5B-ED95-F14F-9A9D-CB69309D2380.root")
`

**Reply by riga** [reply]
Interesting. But is there anything contained in e.g. `f["Events"].keys()`?

I just tried to access the file as well, and I see a "permission denied" error. For any other file in that `/QCD_Pt-170to300_EMEnriched_TuneCP5_13TeV-pythia8/RunIISummer20UL18NanoAODv9-106X_upgrade2018_realistic_v16_L1v1-v2/NANOAODSIM` it works, it's just that particular file that seems to fail, so I'd assume it's not cf related, but rather a rucio thing.

Fyi, I tested with

```bash
gfal-stat root://cms-xrd-global.cern.ch//store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-170to300_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/250000/3861392C-D7EF-E440-BC8E-50D6D4917F25.root
```

which fails, whereas (a different file)

```bash
gfal-stat root://cms-xrd-global.cern.ch//store/mc/RunIISummer20UL18NanoAODv9/QCD_Pt-170to300_EMEnriched_TuneCP5_13TeV-pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v2/2430000/763BDB0F-3D17-A342-B6F4-AA8DA08A6D45.root
```

works. They are stored on different sites, and the list of sites of the file that fails contains some "usual suspects" of sites that often lead to errors (at least that's what we observed). However, it works for me with the EU redirector (xrootd-cms.infn.it), so you should consider adding that one to your list of fs' (config value `[outputs] lfn_sources`) to check too.

**Reply by riga** [reply]
Ah, I just saw that you actually have that one in your list (`wlcg_fs_infn_redirector`). Then I hope it's just a temporary thing. Maybe add the `wlcg_fs_fnal_redirector` as well to the list, I just added the entry to the main law.cfg in the master.
